low-level/framework/platforms/server/platform.py

```diff
@@ -29,32 +30,36 @@ def __init__(self):
         self._ipmi = IpmiFactory().get_implementor("ipmitool")
 
     @staticmethod
-    def get_os():
-        """Returns os name({ID}{VERSION_ID}) from /etc/os-release."""
+    def get_os_info():
```

Comment:
I did some minor changes for it in the platform.py file only.
I did not make change in Utility.py for now because lots of cyclic import dependency is happening if I made the changes.
We need separate PR to take care of these things. I feel that to not make this change in this PR right away because it will affect many files.

Reply:
After that, I have analyzed it. If someone wants hole data instead of a particular field. then he needs to call that function with every attribute.
instead of we can provide full information of os and he will take what will be required out of that.

Reply:
**JFYI:** Version number in `/etc/os-realse` is not in it's full form. the full version number is available in a pretty string at `/etc/{os}-release`.

Reply:
I would recommend using python's platform library and building the required string on our own which won't be OS dependent.

```
def get_os():
    os_info = None
    system = platform.system()
    if system == 'Linux':
        os_info = ' '.join(platform.linux_distribution())
    else:
        os_info = f"{platform.system()} {platform.release()} {platform.version()}"
    return os_info
```
```
>>> get_os() 
CentOS Linux 7.8.2003 Core

>>> get_os()
Rocky Linux 8.4 Green Obsidian

>>> get_os()
Windows 10 10.0.19041
```

Reply:
Yes, @jjxsg I agree with your comment but we need hole system data. not just os name. I know the platform module will provide all this information. but It is not the correct way to write a function to get only os name. we can directly import the platform and get os name from that at run time.
This function is written because only to get full os information. which we will easily get using /etc/os-release.

`[721836@ssc-vm-3499 cortx-monitor]$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
**PRETTY_NAME**="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"`

@sumedhak27 I have tried with /etc/centos-release but it will return only os name. not full information.
`[721836@ssc-vm-3499 cortx-monitor]$ cat /etc/centos-release
CentOS Linux release 7.8.2003 (Core)`

Reply:
Yes @satish-seagate , but the version_id does not mention full version number, which should be `7.8` , we need to extract it from `/etc/{os}-release` file only.

Hi, @jjxsg , `platform.linux_distribution()`  is deprecated from python 3.6 onwards

Reply:
@satish-seagate the function was just an example to demonstrate the idea. My point is, since we are moving to different distros (Rocky Linux and others in future), the approach to get OS details should be OS independent. 
For manifest, we need all other details so current approach is good enough.

Reply:
@sumedhak27 Good catch, we must not use `platform.linux_distribution()`

Reply:
`For changes that are required for Rocky Linux porting this full version is important. so I was redesigning the same function as well.`
@sumedhak27 Okay